### DISCUSSION TITLE
Add CI workflow for Windows 🚀

### DISCUSSION THREAD

**Discussion by lucydodo** [sqlitebrowser/sqlitebrowser#3508] [Show and tell]
We just added the CI workflow for Windows.

On the development side: we can check the code for reliability.
On the end user side: Continuous builds for Windows environments are now available.

You can download the continuous build from https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/continuous
If you have any feedback, we'd love to hear it. Thank you.

**Comment by justinclift** [comment]
AWESOME!  Super well done @lucydodo. :grin:

Looking at the `continuous` release now, this is what's showing up:

![image](https://github.com/sqlitebrowser/sqlitebrowser/assets/406299/efc7bf12-4d70-4601-9a98-713702372d78)

How feasible is it to?

1. Add the git commit id into the filename like with the others (.dmg, .AppImage)
2. Add a .zip version, so people can unzip and run it easily.  No need for admin level permission.

@chrisjlocke Pretty sure you'll want to try them out... :smile:

**Reply by lucydodo** [reply]
Thanks, and to be honest, this wouldn't have happened without your help. and I've taken your feedback on board and am currently testing it in the testing repo. I'll push it upstream as soon as I'm done testing it. 😄

**Reply by lucydodo** [reply]
![Screenshot 2023-11-19 at 06 48 20](https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/8d6bd4c1-4823-45f2-b14d-322ee718e8c1)
I've made changes based on your feedback. 😄

**Reply by justinclift** [reply]
Cool, the git commit is now in the filename :+1:, and there's a .zip file version as well for the people who need that :+1:.

But, er... what happened to the .msi ones?  We still need those too. :grin:

**Reply by lucydodo** [reply]
Oh, I thought you wanted a replacement. I'll add MSI back in as well. 😄

**Reply by lucydodo** [reply]
![Screenshot 2023-11-19 at 17 55 13](https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/90a49924-29aa-4e52-b988-b14f7a2d385c)
Now we have MSI files too 😄

**Reply by lucydodo** [reply]
In the long run, we'll be able to add notarization to the CI pipeline, just as we do with macOS CI work.
and perhaps even replace the existing nightly build system.

**Reply by justinclift** [reply]
Replacing the existing nightly build system would be a welcome thing. :+1:

Signing of windows binaries seems a lot more complicated than the macOS ones.  There's no real centralised certificate issuer for Windows like Apple provides.

Instead there are completing companies that offer "enterprise signing" certs.  It's widely regarded (by developers) as a scam, as it can cost several hundred $ (per year) just for a certificate that means windows *mostly* won't display a warning to the user.  But sometimes still does anyway. :face_exhaling:

**Reply by lucydodo** [reply]
Cool, I'll be emailing you about this later today. 😄
